src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -952,14 +951,30 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                         for (i = 0; (uint)i < maxIters; i++)
                         {
                             utf16Data = Unsafe.ReadUnaligned<Vector128<short>>(pInputBuffer);
-                            if (!Sse41.TestZ(utf16Data, nonAsciiUtf16DataMask))
+
+                            if (AdvSimd.IsSupported)
                             {
-                                goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
-                            }
+                                Vector128<short> isUtf16DataNonAscii = AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask);
+                                bool hasNonAsciiDataInVector = AdvSimd.Arm64.MinAcross(isUtf16DataNonAscii).ToScalar() != 0;
```

Comment:
>MinAcross [](start = 77, length = 9)

no `MinAcross` :)

Reply:
Ah, just noticed @echesakovMSFT comment https://github.com/dotnet/runtime/pull/39041#discussion_r454731684. The alternative approach without using `MinAcross` could be the following that is done in https://github.com/dotnet/runtime/pull/39103.

```csharp
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static bool ContainsNonAsciiValue(Vector128<ushort> value)
        {
            if (!AdvSimd.Arm64.IsSupported)
            {
                throw new PlatformNotSupportedException();
            }
            value = AdvSimd.Arm64.MaxPairwise(value, value);
            return (value.AsUInt64().ToScalar() & 0x8000800080008000) != 0;
        }
```

Reply:
Sure, this could be done without using MinAcross() instruction
```c#
Vector128<ushort> isUtf16DataNonAscii = AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask);
bool hasNonAsciiDataInVector = AdvSimd.Arm64.MaxPairwise(isUtf16DataNonAscii, isUtf16DataNonAscii).AsUInt64().ToScalar() != 0;

if (hasNonAsciiDataInVector)
{
    goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
}
```

@kunalspathak Just to clarify, why are we avoiding Across() instructions?

Reply:
Thanks! For this specific case, the `CompareTest` overload I have to use is the one that returns a `Vector128<short>`. It's the only thing I will use differently to that suggestion.

Reply:
@echesakovMSFT - See [this](https://github.com/dotnet/runtime/pull/38707#discussion_r454400897) and [this](https://github.com/dotnet/runtime/pull/38653#discussion_r454419458) explanation from Tamar.

Reply:
Also, just noticing @echesakovMSFT suggestion, you also need the ` & 0x8000800080008000` on the result of `.ToScalar()` to check if MSB was set and if yes, then it is non-ascii.

Reply:
@kunalspathak Can you please clarify why do we need to check MSBs?

Would not it be enough to check that all `isUtf16DataNonAscii [2*i] + isUtf16DataNonAscii [2*i+1]` for `i` in `[0,1,2,3]`, are zero?
If any of `isUtf16DataNonAscii[i]` is non-zero (i.e. `= 0xFFFF`) then the corresponding sum would not be zero as well and `MaxPairwise()[0] != 0` will return true

Am I missing something?

Reply:
Actually you are right @echesakovMSFT . I got confused with the input on which we operate. Unlike done in #39103 where the input is saturated using `0x7F80` before checking if it contains non-ascii, here, we directly operate on the data. So in that case, `.AsUInt64().ToScalar() != 0` should get the right value.